packages/core/admin/admin/src/content-manager/pages/EditView/components/FormInputs/DynamicZone/Field.tsx

```diff
@@ -58,10 +58,12 @@ const DynamicZone = ({
     })
   );
 
-  const { value = [], error } =
-    useField<Array<Attribute.GetValue<Attribute.DynamicZone>[number] & { __temp_key__: number }>>(
-      name
-    );
+  type DzWithTempKey =
```

Comment:
Maybe it's worth adding a comment to clarify this a little?

Reply:
What kind of clarification would you add?

Reply:
The temp key is used in the CM to enable the drag and drop in array-like structures (toMany relations, repeatable components, dynamic-zones, etc...).

Here we just define the type that reflects this CM behavior. I can add a comment here (I don't mind), but it means that we need to document every other instance of `__temp_key__` to be consistent.

Reply:
Just putting my opinion here, I don't think we need to put lots of comments explaining these types of behaviors in comments, let's leave this to when we plan something better in the contributor documentation instead